### ISSUE TITLE
Allow Model Elements (e.g., Classes, Tables, Entities) to Be Treated as Namespaces

### ISSUE THREAD

**Issue by stojimirovic** [AelasticS/aelastics#7]
**Description**:

Currently, in **aelastics-types** package, it is required that all names of model elements  within a model or namespace must be unique. As a result, it is not possible to have multiple elements with the same name in different contexts. For example:
- You cannot have columns with the same name in two different relational tables.
- Similarly, classes and properties in object-oriented (OO) models, or attributes for entities in an entity-relationship (ER) model must all have unique names.

**Proposal**:

To resolve this limitation, we propose that certain model elements, such as classes, tables, and entities, should be treated as namespaces. This would allow elements with the same name to coexist in different contexts (e.g., the same column name in different tables, or the same class name in different namespaces).

To implement this change, we suggest defining these concepts (classes, tables, entities) as **_intersection types_** between a model element and a namespace. This would enable the desired behavior while maintaining the overall integrity of the schema.

**Impact**:

- Existing metamodels and models in the project would need to be updated to accommodate this change.
- Adjustments to model validation and parsing logic will be necessary to handle model elements defined as intersection types.

**Next Steps**:

1. Modify the metamodels to allow classes, tables, and entities to function as namespaces.
2. Update relevant validation rules to account for intersection types.
3. Refactor models within the project to ensure compatibility with the new namespace functionality.
4. Update JSX rendering logic if needed and test. 

This change should help support more flexible modeling patterns, particularly for complex systems that require reusability of names across different contexts.

**Comment by Sinisa-Neskovic** [comment]
I agree completely. But this can be implemented only after issue #4 is completed.
